arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -354,6 +360,47 @@ public void setBatchSize(int batchSize) {
     }
   }
 
+  private static final class PositionVectorReader extends VectorizedArrowReader {
+    private long rowStart;
+    private NullabilityHolder nulls;
+
+    @Override
+    public VectorHolder read(VectorHolder reuse, int numValsToRead) {
+      Field arrowField = ArrowSchemaUtil.convert(MetadataColumns.ROW_POSITION);
+      FieldVector vec = arrowField.createVector(ArrowAllocation.rootAllocator());
+
+      if (reuse != null) {
+        vec.setValueCount(0);
+        nulls.reset();
+      } else {
+        ((BigIntVector) vec).allocateNew(numValsToRead);
+        for (int i = 0; i < numValsToRead; i += 1) {
+          vec.getDataBuffer().setLong(i * Long.BYTES, rowStart + i);
+          nulls = new NullabilityHolder(numValsToRead);
```

Comment:
Why are we setting this inside of the for loop?

Reply:
Thanks @holdenk, This is a problem. Let me move this out of the loop.